main.cpp

```diff
@@ -127,596 +161,485 @@ struct CarWash
+    return output;
+}
+bool SwissArmyKnife::openABottle(double speedInDegreesPerSecond, bool isACorkLid)
+{    
+    if (speedInDegreesPerSecond > 20)
+    {
+         return false;
+    } else if (isACorkLid)
+    {
+        return true;
+    }
+    return false;
+}
+void SwissArmyKnife::unscrew(Tool toolB)
+{    
+    std::to_string(toolB.lentghInCm);
+}
 
-struct Speakers //                            1) define an empty struct for each of your 10 types.       
+struct Speakers    
 {
-    //Amount of devices                   2) copied and commented-out plain-english property
-    int amountOfDevices = 4;  //                   3) member variables with relevant data types.
-    //Size in cm   
+    int amountOfDevices = 4; 
     int sizeInCm = 1;     
-    //Maximum power in watts (double)   
     double maxPowerWatts= 5.0;              
-    //Maximum decibels capacity (double)         
     double maxDecibPower = 70;               
-    //Number of wirings of the coil (int)            
     int numberOfcoilwirings = 200;               
     
     struct SoundFile      
     {
-        //2) member variables with relevant data types.  the names are appropriate for the U.D.T.'s purpose.
         float sampleRateInKhz = 44.0f;
         float soundLengthInMs = 3000.0f;   
         std::string fileFormat = "wav";
         std::string filePath = "/root/ding.wav";
 
-        //3) a member function.  it has multiple arguments, some with default values.
-        //the parameter names are related to the work the function will perform.
-        // returns true if contains frequencies exceeding 20Khz
         bool verifyIfIsUltrasound();
-        // returns compression ratio achieved
         float compressFile(int compressionLevel = 3);
-        // 
         void applyBandPassFilter(double freqA, double freqB);
     };
 
-    //3) a member function.  it has multiple arguments, some with default values.
-    //the parameter names are related to the work the function will perform.
-    // returns seconds taken to play the sound
     float playAudibleSound(SoundFile soundFileA, double level = 6.0);
-    // returns if noise reduction already active
     bool activateNoiseReduction(int duration);
-    // returns seconds taken to play the sound
     float playUltraSound(SoundFile soundFileB, double level);
     
-    //5) a member variable whose type is a UDT.
     SoundFile soundFileBeingPlayed;
 };
 
-/*
-Thing 6) Cables
-5 properties:
-    1) Length in cm (int)
-    2) Color (std::string)
-    3) Amount of internal wires (int)
-    4) External material (std::string)
-    5) Internal wire thickness in um (int)
-3 things it can do:
-    1) Hold the device over user's head
-    2) Transport audio signals
-    3) Transport control signals
- */
+float Speakers::playAudibleSound(SoundFile soundFileA, double level)
+{    
+    float output = soundFileA.soundLengthInMs * static_cast<float>(level);
+    return output;
+}
```

Comment:
Casting has not been introduced. Do not use casting. Change `level` into a `float`
- [ ] revise